### ISSUE TITLE
Why two columns but not three?

### ISSUE THREAD

**Issue by burnce** [jgm/pandoc#5258]
Hello,

It is trivial to modify the "Two Content" layout in a powerpoint template to actually have three columns. The "Outline" view in PowerPoint will correctly create 3 labeled bullets -- one for each column. Unfortunately Pandoc will only populate the first two columns. It would be cool it the number of actual columns was extendable.

Extract from my in.utf8.md file:
````
### Columns

:::::::::::::: {.columns}
::: {.column}
contents 1
:::
::: {.column}
contents 2
:::
::: {.column}
contents 3
:::
::::::::::::::
````

Pandoc command:
````
pandoc +RTS -K512m -RTS in.utf8.md --to pptx --from markdown+autolink_bare_uris+ascii_identifiers+tex_math_single_backslash+smart --output in.pptx --reference-doc custom-reference.pptx
````

Cheers,

B.

**Comment by jkr** [comment]
The issue is that we can only make use of layouts that we know will already be there -- the two-column layout is included in the templates included with word, and most other templates. Since layout in pptx is dependent on the pre-existing layouts, it isn't something that can be easily extended programatically.

If we did include our own extended layout, then it would break when people tried to switch to their own templates, or specify them on the command line. As with most things in PowerPoint, if you want to do something ad-hoc or non-standard, you're better off doing it by hand.

That being said, there *should* be a warning. I'll leave this issue open until I check on that.

**Comment by little-big-h** [comment]
I was wondering why this couldn't be done in a more generic way in general. It is possible to assign names/IDs to elements in powerpoints. Couldn't those be used to identify placeholders match their IDs to div-elements in the input and fill-in their content?

**Comment by jkr** [comment]
Unfortunately, every different number of columns would require its own layout slide -- that's just how PPT works. Two-column comes by default in most ms-supplied templates, and therefore most that are based on them. We could make our own three-column layout, but it would only work with the template we supply.
________________________________

**Comment by little-big-h** [comment]
Right, that makes sense. To be honest, I am, personally, not interested in a three-column slide but in an extensible mechanism to use arbitrary user-provided templates. I think this would definitely fall in the "power-user feature" category but as things stand right now, beamer gives me a lot of flexibility where powerpoint gives me virtually none. A simple "fill in the placeholders based on ID" approach should cover a lot of ground there.

Frameworks like D3 get a lot of mileage out of the approach.

**Comment by ickc** [comment]
Off topic: this is true for images or mixing image with text, etc. I suppose it is related to the template slide limitation you mentioned. It seems it is much harder to solve than I thought. And it is a huge limitation on the pptx writer comparing to other slide writer in pandoc.

I think it would be useful to document its limitation. I stated my own unofficial doc to test what features is supported in pptx writer: <https://github.com/ickc/pandoc-pptx>.

Just asking as someone who don’t know the details: how hard would it be to automatically generate a 3 column template out of existing 2 columns one?
